src/spaces.hpp

```diff
@@ -248,6 +248,14 @@ struct vector : entity<vector<T, N>> {
 template <class... Ts>
 vector(Ts...) -> vector<std::common_type_t<Ts...>, sizeof...(Ts)>;
 
+template <class, std::size_t>
+struct extend_to {};
+
+template <Arithmetic T, std::size_t N, std::size_t NewN>
+struct extend_to<vector<T, N>, NewN> {
```

Comment:
What does this do?

Reply:
The name is poor. I use it to generate vectors of the same "kind" but with different dimensions, which is used to reshape matrices. The way matrices depend on vectors is poor, though